homeassistant/components/sensor/deconz.py

```diff
@@ -0,0 +1,150 @@
+"""Platform integrating Deconz sensor support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor/deconz/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components.deconz import (
+    CONF_TYPE_AS_EVENT, DECONZ_DATA, DOMAIN, DeconzEvent)
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.icon import icon_for_battery_level
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup sensor platform for Deconz."""
+    from pydeconz.sensor import DECONZ_SENSOR
+
+    type_as_event = discovery_info.get(CONF_TYPE_AS_EVENT)
+
+    if DECONZ_DATA in hass.data:
+        sensors = hass.data[DECONZ_DATA].sensors
+
+    for _, sensor in sensors.items():
+        if sensor.type in DECONZ_SENSOR:
+            if sensor.type in type_as_event:
+                DeconzEvent(hass, sensor)
+                if sensor.battery:
+                    async_add_devices([DeconzBattery(sensor)], True)
+            else:
+                async_add_devices([DeconzSensor(sensor)], True)
+
+
+class DeconzSensor(Entity):
+    """Representation of a sensor."""
+
+    def __init__(self, sensor):
+        """Setup sensor and add update callback to get data from websocket."""
+        self._sensor = sensor
+        self._sensor.register_callback(self._update_callback)
+
+    @callback
+    def _update_callback(self):
+        """Update the sensor's state, if needed."""
+        self.async_schedule_update_ha_state()
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._sensor.state
+
+    @property
+    def name(self):
+        """Return the name of the event."""
+        return self._sensor.name
+
+    @property
+    def device_class(self):
+        """Class of the event."""
+        return self._sensor.sensor_class
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return self._sensor.sensor_icon
+
+    @property
+    def unit_of_measurement(self):
+        """Unit of measurement of this entity."""
+        return self._sensor.sensor_unit
```

Comment:
I think you should translate here into home assistant - standardized units such as `TEMP_CELSIUS` (which is defined as `'C'`, while pydeconz would use `'Celsius'`). The same probably holds for other units?

Reply:
Good catch! Too quick on that one.